server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -366,8 +390,18 @@ public AllocationStatus getLastAllocationStatus() {
      *
      * @return calculated delay in nanoseconds
      */
-    public long getRemainingDelay(final long nanoTimeNow, final Settings indexSettings) {
-        long delayTimeoutNanos = INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.get(indexSettings).nanos();
+    public long getRemainingDelay(
+        final long nanoTimeNow,
+        final Settings indexSettings,
+        final Map<String, SingleNodeShutdownMetadata> nodesShutdownMap
+    ) {
+        Map<String, SingleNodeShutdownMetadata> nodeShutdowns = nodesShutdownMap != null ? nodesShutdownMap : Collections.emptyMap();
```

Comment:
I don't think we need this `null` check here, since `Metadata#nodeShutdowns()` returns an empty map if there are no shutdowns. Maybe we can replace it with an `assert` instead?

Reply:
👍 I think this is left over from an intermediate version.